xl2times/__main__.py

```diff
@@ -30,22 +45,29 @@ def _read_xlsx_cached(filename: str) -> list[EmbeddedXlTable]:
     Since excel.extract_tables is quite slow, we cache its results in `cache_dir`.
     Each file is named by the hash of the contents of an xlsx file, and contains
     a tuple (filename, modified timestamp, [EmbeddedXlTable]).
+
+    Args:
+        filename: Path to the xlsx file to extract tables from.
     """
     with open(filename, "rb") as f:
         digest = hashlib.file_digest(f, "sha256")  # pyright: ignore
     hsh = digest.hexdigest()
-    if os.path.isfile(cache_dir + hsh):
-        fname1, _timestamp, tables = pickle.load(open(cache_dir + hsh, "rb"))
+    hash_file = cache_dir / f"{Path(filename).stem}_{hsh}.pkl"
```

Comment:
If the cache filename contains the original filename, then perhaps we no longer need to store a tuple in the pickle? And as discussed we probably don't need to check modified timestamp, since it's super unlikely that files get modified but their hash remains the same..